src/status_im/ui/screens/contact_requests/views/contact_request.cljs

```diff
@@ -0,0 +1,165 @@
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.screens.contact-requests.styles :as styles]
+            [status-im.utils.handlers :refer [<sub]]
+            [status-im.ui.screens.chat.photos :as photos]
+            [status-im.multiaccounts.core :as multiaccounts]
+            [status-im.ui.components.icons.icons :as icons]
+            [status-im.utils.contenthash :as contenthash]
+            [status-im.constants :as constants]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.screens.home.views.inner-item :as home-item]))
+
+(defn mention-element [from]
+  (str "@" @(re-frame/subscribe [:contacts/contact-name-by-identity from])))
+
+(def max-contact-request-length 160)
+(def max-contact-request-lines 2)
+
+(defn add-parsed-to-message [acc {:keys [type destination literal children]}]
+  (let [result (case type
+                 "paragraph"
+                 (reduce
+                  (fn [{:keys [_ length] :as acc-paragraph} parsed-child]
+                    (if (>= length max-contact-request-length)
+                      (reduced acc-paragraph)
+                      (add-parsed-to-message acc-paragraph parsed-child)))
+                  {:components [quo/text]
+                   :length     0}
+                  children)
+
+                 "mention"
+                 {:components [quo/text {:style styles/mention-text} [mention-element literal]]
+                  :length     4}                            ;; we can't predict name length so take the smallest possible
+
+                 "status-tag"
+                 (home-item/truncate-literal (str "#" literal))
+
+                 "link"
+                 (home-item/truncate-literal destination)
+
+                 (home-item/truncate-literal literal))]
+    {:components (conj (:components acc) (:components result))
+     :length     (+ (:length acc) (:length result))}))
+
+(defn message-wrapper
+  ([] (message-wrapper 1))
+  ([number-of-lines]
+   [react/text-class {:style               styles/contact-request-message-text
+                      :number-of-lines     number-of-lines
+                      :ellipsize-mode      :tail
+                      :accessibility-label :chat-message-text}]))
+
+(defn render-contact-request-message
+  "Render the preview of a contact-request message to a maximum of max-length characters"
+  ([parsed-text] (render-contact-request-message parsed-text max-contact-request-length 1))
+  ([parsed-text max-length number-of-lines]
+   (let [result
+         (reduce
+          (fn [{:keys [_ length] :as acc-text} new-text-chunk]
+            (if (>= length max-length)
+              (reduced acc-text)
+              (add-parsed-to-message acc-text new-text-chunk)))
+          {:components (message-wrapper number-of-lines)
+           :length     0}
+          parsed-text)]
+     (:components result))))
+
+(defn message-content-text [{:keys [content content-type community-id]}]
```

Comment:
this function seems to be duplicated in a few places ?

Reply:
this code is not needed anymore actually, removing the whole directory